### PR TITLE
Feature/members only

### PR DESCRIPTION
**Pull request by jpdf00**
In this project, we implemented authentication systems so users can only access areas of a site they are authorized to.

- If no user is logged, the index page shows Sign up(for new users), login (for registered users), and a list of posts with "Anonymous" as the post creator.
- If the user is logged, the index page shows the users name(with a link for editing), logout (to end the session), new post(with a link to create a new post), and a list of posts with the name and username of their respective creators.

### CODE REVIEW

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
## Status: Changes Required! :recycle:
### Hi @jpdf00 :clap: :clap:
#### You did an awesome job, you deserve :+1: for what you have did so far. However, there are some requirements missing/not implemented correctly. Check my comments below :point_down:

#### PostController:
 - it should use `before_action` to restrict access to the `:new` and `:create` methods. In this case the `:new` method is missing in the `before_action` filter.
 - `:new` should initialize an empty post `@post = Post.new`

#### _Reviewed by Temitayo Adeleke_

File: app/views/layouts/application.html.erb [hunk 1]
```diff
@@ -11,6 +11,34 @@
...
+        <%= link_to 'New Post', new_post_path, class: "button is-info is-inverted" %>
+      </p>
+    <% else %>
+      <p>
+        <%= link_to 'Log In', new_user_session_path, class: "button is-info" %>
+      </p>
+      <p>
+        <%= link_to 'Sign Up', new_user_registration_path, class: "button is-info" %>
+      </p>
+    <% end %>
```
Comment (acushlakoncept) [inline app/views/layouts/application.html.erb]: There should be absolutely NO logic in views.

File: app/views/posts/_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    </div>
+  <% end %>
+
+  <%= form.label :body, "Write new post below:" %><br>
+  <%= form.text_area :body, placeholder: "texttt", autofocus: true %><br>
+
+  <div class="actions">
+    <%= form.submit %>
+  </div>
+<% end %>
```
Comment (acushlakoncept) [inline app/views/posts/_form.html.erb]: There should be absolutely NO logic in views.

File: app/views/posts/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+      <% if user_signed_in? %>
+        <strong><%= post.user.name %></strong><br>
+        <small>@<%=  post.user.username %></small><br>
+      <% else %>
+        <strong>Anonymous</strong><br>
+      <% end %>
+      <p><%= post.body %></p>
+    </li><br>
+  <% end %>
+</ul>
```
Comment (acushlakoncept) [inline app/views/posts/index.html.erb]: There should be absolutely NO logic in views.

**Review by acushlakoncept — CHANGES_REQUESTED** [review]
# Status: Changes Required ♻️

### Hello @jpdf00 and @anapdh , Great job! 👏  Both of you have done well on this project.

However, there are still some changes that are necessary for this project to be approved.
Let us quickly add some finishing touches and request another review.

- [x] Add validations to your post model, I should not be able to create an empty post

- [x] There should be absolutely NO logic in views. Views are only for displaying information that became available to it from its corresponding controller not generating it.  ([screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/rails/section/no_logic_in_helpers_requirement.png) from Pathwright)
I have gone ahead and made some comments on your codebase see affected files below:

**Don't know where to start or how to?** See  [Rails View Helpers Best Practices](https://medium.com/@acushla4real/rails-view-helpers-best-practices-guide-d7d7b6db3db4) 

- [x] Do not forget to add `yarn` to your getting started instructions, perfect after the `bundle install`. I couldn't run the project without running `yarn`

### Cheers and Happy coding! 💻 👏

(Feel free to contact me on slack(@acushla) or reply to this thread, if you have any additional question)
Uduak Essien
[Twitter](https://twitter.com/acushlakoncept) | [LinkedIn](https://www.linkedin.com/in/acushlakoncept/)

**Review by MahmoudBakr23 — CHANGES_REQUESTED** [review]
# **Changes Required 🛑♻**

<hr>

Hello there, great job you've done so far on this project and on the required changes. You refactored the logic of your views, put the before_action in perfect usage. So, well done. However, there are still a few more fixes to make in order to get it completely perfect :)

- [x] Since you're limiting the posts route resources to only index, new, and create, there's no need to show, edit, update, and destroy methods in the controller. They're not even working! Please remove them and their related code such as the set_post method from your controller and view related files.
posts_routes_requirement

- [x] Your readme file misses the ```rails db:create``` before the ```rails db:migrate``` CLI

<hr>

Kindly make the required changes, then ask for another review.
Good luck and happy coding 👨‍💻💻

**Review by ghost — APPROVED** [review]
# Status: Approved 🎉 

**Congratulations team 👏. Good luck with your next project 😃.**

**Pull request merged**